src/libxtp/threecenter_gwbse.cc

```diff
@@ -77,8 +101,24 @@ void TCMatrix_gwbse::Fill(const AOBasis& gwbasis, const AOBasis& dftbasis,
     }
     // Fill block for this shell (3-center overlap with _dft_basis +
     // multiplication with _dft_orbitals )
-    FillBlock(block, shell, dftbasis, dft_orbitals);
+    std::vector<Eigen::MatrixXd> symmstorage =
+        ComputeSymmStorage(shell, dftbasis, dft_orbitals);
 
+    // If cuda is enable each OpenMP Perform the convolution in the cuda device.
+    // Each thread has its own stream that behaves like a queue containing the
+    // matrix multiplication requests. If there are more threads than
+    // MAXIMUM_GPU_STREAMS, the remaining threads will perform the perform the
+    // convolution using the default method.
+#if defined(USE_CUDA)
+    if (OPENMP::getThreadId() < _max_gpu_streams) {
```

Comment:
nice, can we do sth. similar for the stuff above?

Reply:
done